src/sentry_core.c

```diff
@@ -511,6 +500,60 @@ sentry__prepare_event(const sentry_options_t *options, sentry_value_t event,
     return NULL;
 }
 
+sentry_envelope_t *
+sentry__prepare_transaction(const sentry_options_t *options,
+    sentry_value_t transaction, sentry_uuid_t *event_id)
+{
+    sentry_envelope_t *envelope = NULL;
+
+    bool should_skip = !sentry_value_is_true(
+        sentry_value_get_by_key(transaction, "sampled"));
+    if (should_skip) {
+        SENTRY_DEBUG("throwing away transaction due to sample rate");
```

Comment:
nitpicking, but it is my understanding that this is not always due to the sample rate as you can explicitly sample-or-not a transaction overriding the sample rate?

Reply:
true that, i'll add that as a possible cause in here to avoid misleading anybody.